lib/berkshelf/middleware/follow_redirects.rb

```diff
@@ -0,0 +1,135 @@
+require "set"
+require "berkshelf/mixin/logging"
+
+module Berkshelf
+  module Middleware
+    # Borrowed and modified from:
+    # {https://github.com/lostisland/faraday_middleware/blob/master/lib/faraday_middleware/response/follow_redirects.rb}
+    #
+    # Public: Follow HTTP 301, 302, 303, and 307 redirects for GET, PATCH, POST,
+    # PUT, and DELETE requests.
+    #
+    # This middleware does not follow the HTTP specification for HTTP 302, by
+    # default, in that it follows the improper implementation used by most major
+    # web browsers which forces the redirected request to become a GET request
+    # regardless of the original request method.
```

Comment:
Should we just force this behavior on? I can't think of a case in the Chef world where switching to GET is what we want.